### DISCUSSION TITLE
mcuboot Secondary slot on external SPI flash- frdm k64f

### DISCUSSION THREAD

**Discussion by jahedul-haque** [mcu-tools/mcuboot#1595] [General]
Hello Team,

I am using FRDM K64F Board , I have attached an External SPI Flash flash to the board and have been able to read write successfully with Zephyr RTOS. Now I want to move the mcuboot secondary slot from flash to external SI flash , I have tried with with overlays and config but none of them seems to be working. Could you please provide an example where external spi flash has been used for secondary slot , all examples I found online are for nordic boards which I have tried with K64f and was not working. 
so, the configuration will be mcuboot and primary slot on internal flash and secondary flash on external spi flash. I am using overwrite mode so scratch partition is not used. I need to use Zephyr RTOS with mcuboot.

Thanks ,
Jahedul Haque

**Comment by de-nordic** [comment]
You need to use the same overlays defining the SPI device and flash partitions as you use for your application.
If you are building MCUboot as a Zephyr application, then your MCUboot should have the same options enabling SPI access as your application.
